x/ccv/provider/keeper/throttle.go

```diff
@@ -494,3 +494,54 @@ func (k Keeper) SetLastSlashMeterFullTime(ctx sdktypes.Context, time time.Time)
 	store := ctx.KVStore(k.storeKey)
 	store.Set(providertypes.LastSlashMeterFullTimeKey(), sdktypes.FormatTimeBytes(time.UTC()))
 }
+
+// GetThrottledPacketData fetches packet data from the store using consumerChainId and ibcSeqNum
+// Since multiple types can be stored, it is up to the caller to determine the type after fetching
+func (k Keeper) GetThrottledPacketData(ctx sdktypes.Context, consumerChainID string, ibcSeqNum uint64) interface{} {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(providertypes.ThrottledPacketDataKey(consumerChainID, ibcSeqNum))
+
+	var packetData interface{}
+	var err error
+	switch bz[0] {
+	case slashPacketData:
+		spd := ccvtypes.SlashPacketData{}
+		err = spd.Unmarshal(bz[1:])
+		packetData = spd
+	case vscMaturedPacketData:
+		vpd := ccvtypes.VSCMaturedPacketData{}
+		err = vpd.Unmarshal(bz[1:])
+		packetData = vpd
+	default:
+		panic("invalid packet data type")
+	}
+
+	if err != nil {
+		panic(fmt.Sprintf("failed to unmarshal pending packet data: %v", err))
+	}
+
+	return packetData
+}
+
+// GetSlashPacketForGlobalEntry fetches a slash packet data from the store using consumerChainId and ibcSeqNum
+// If the packets is not SlashPacketData, it is considered as not found.
+func (k Keeper) GetSlashPacketForGlobalEntry(ctx sdktypes.Context, consumerChainID string, ibcSeqNum uint64) (ccvtypes.SlashPacketData, bool) {
```

Comment:
If this is only neccessary for queries I'd prefer that it lives in a query file, unless that's not possible due to package stuff (that's the case for test packages and `GetAllThrottledPacketData`)